docs/content/guides/dagster/validating_pandas_dataframes_with_dagster_types.mdx

```diff
@@ -0,0 +1,240 @@
+
+<Image
+alt="data.png"
+src="/images/guides/validating_pandas_dataframes_with_dagster_types/data.png"
+width={730}
+height={608}
+/>
+
+The first thing you want is to visualize the distribution of trip lengths, so you set up a simple Dagster job to do this. Your job has two ops: `load_trips` reads a CSV with your data, and `plot_trip_length_dist` generates a plot of the distribution using the [Holoviews](https://holoviews.org/) plotting library.
+
+```python file=/guides/validating_pandas_dataframes_with_dagster_types/job_1.py
+from dagster import AssetMaterialization, job, op
+import numpy as np
+import pandas as pd
+import holoviews as hv
+
+@op
+def load_trips():
+    return pd.read_csv(
+        "./2021-10_ebike_trips.csv",
+        parse_dates=["start_time", "end_time"],
+    )
+
+@op
+def generate_plots(trips):
+    minute_lengths = [ x.total_seconds() / 60 for x in trips.end_time - trips.start_time ]
+    edges, freqs = np.histogram(minute_lengths, 15)
+    hist = hv.Histogram((freqs, edges)).opts(width=600, xlabel='Minutes')
+    hv.save(hist, 'trip_length_hist.png', fmt='png')
+    yield AssetMaterialization(
+        asset_key="trip_dist_plot",
+        description="Distribution of trip lengths."
+    )
+    yield Output(None)
+
+@job
+def generate_trip_plots():
+    generate_plots(load_trips)
+```
+
+If we try running this job and inspect the result, we get a strange error:
+
+```
+# => ...
+# => ValueError: autodetected range of [nan, nan] is not finite
+```
+
+## Schema
+
+Since our code looks OK, the problem likely originates from the data. To investigate, we write a [Pandera schema](https://pandera.readthedocs.io/en/stable/dataframe_schemas.html) and try to validate our data. Our schema is, like our dataset, very simple. It asserts column data type and existence. Pandera columns are non-nullable by default, so there is also an implicit assertion that all values are defined (NOTE: this illustrates only a small slice of Pandera's functionality):
+
+```python file=/guides/validating_pandas_dataframes_with_dagster_types/schema.py
+import numpy as np
+import pandera as pa
+import pandas as pd
+
+MIN_DATE = pd.Timestamp('2021-10-01')
+
+df = pd.read_csv(
+    "./2021-10_ebike_trips.csv",
+    parse_dates=["start_time", "end_time"],
+)
+
+schema = pa.DataFrameSchema(
+    columns = {
+        "bike_id": pa.Column(int, checks=pa.Check.ge(0)),  # ge: greater than or equal to
+        "start_time": pa.Column(pd.Timestamp, checks=pa.Check.ge(MIN_DATE)),
+        "end_time": pa.Column(pd.Timestamp, checks=pa.Check.ge(MIN_DATE)),
+    },
+)
+
+schema.validate(df)
+# => SchemaError: non-nullable series 'end_time' contains null values:
+# => 22   NaT
+# => 43   NaT
+```
+
+Our call to `schema.validate` has found the culprit. Somehow, null values have crept into our `end_time` column-- maybe some customers never returned their bikes...
+
+## Dagster Type factory
```

Comment:
cont. [1]: then we could introduce Pandera here - say something like we recommend using some sophisticated validation logics and then describe Pandera briefly.